app/src/main/java/protect/card_locker/Utils.java

```diff
@@ -669,21 +670,32 @@ public static void patchColors(AppCompatActivity activity) {
     // rendering mess
     // use after views are inflated
     public static void postPatchColors(AppCompatActivity activity) {
-        TypedValue typedValue = new TypedValue();
-        activity.getTheme().resolveAttribute(android.R.attr.colorBackground, typedValue, true);
-        activity.findViewById(android.R.id.content).setBackgroundColor(typedValue.data);
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) {
+            activity.findViewById(android.R.id.content).setBackgroundColor(resolveBackgroundColor(activity));
+        }
+    }
 
-        if (Build.VERSION.SDK_INT >= 27) {
-            Window window = activity.getWindow();
+    // Either pass an Activity on which to call getWindow() or an existing Window (may be null) returned by that function.
+    public static void setNavigationBarColor(@Nullable AppCompatActivity activity, @Nullable Window window, int color, boolean useLightBars) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
```

Comment:
We really need to start being consistent here and either use numbers or version codes, not both randomly.

I think most of the codebase uses ints but I have not strong opinion except for consistency.

Reply:
I can never remember what Android version or code the numbers map to. At least with `Build.VERSION_CODES.O_MR1` I get a pop-up in Android Studio telling me that's "Android 8.1" and its value is `27`, giving me the version code, version, and level all at once. That's much nicer than hardcoding `27` as I would need to manually look up which android version & code that is. Of course we could add comments for that instead if we really want to hardcode versions, but if we change those we might forget to keep the comment in sync.

I do agree on consistency. And when I looked at the existing codebase I found more version code constants than hardcoded numbers. Happy to follow up with a PR to use version codes in the two remaining places as well.

```
$ ack Build.VERSION.SDK_INT
app/src/main/java/protect/card_locker/CardsOnPowerScreenService.java
143:        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R) {

app/src/main/java/protect/card_locker/CatimaAppCompatActivity.java
37:            if (Build.VERSION.SDK_INT >= 23) {

app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java
206:    // store system locale for Build.VERSION.SDK_INT < Build.VERSION_CODES.N
468:                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {

app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java
582:                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
637:        if (window != null && Build.VERSION.SDK_INT >= 27) {
1064:            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
1098:            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {

app/src/main/java/protect/card_locker/PermissionUtils.java
23:        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
38:        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {

app/src/main/java/protect/card_locker/UCropWrapper.java
33:        if (Build.VERSION.SDK_INT >= 23) {

app/src/main/java/protect/card_locker/Utils.java
205:        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
487:        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
505:        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
676:        if (Build.VERSION.SDK_INT >= 27) {
765:            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {

app/src/main/java/protect/card_locker/contentprovider/CardsContentProvider.java
79:        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {

app/src/main/java/protect/card_locker/preferences/SettingsActivity.java
155:            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
181:                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
195:            contentProviderReadPreference.setVisible(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
```

Reply:
Okay those are very good argument in favour of always using the codes :)